### PR TITLE
feat(angular): Add country code select to phone inputs

### PR DESCRIPTION
**Pull request by wlee221**
*Issue #, if available:*

*Description of changes:* This adds country code to sign in and sign up. Tested separately with sign-in and with sign-up. There is an auto signin with phone number issue with all frameworks, which is being tracked in #404 and will be addressed in another PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/angular/projects/ui-angular/src/lib/primitives/amplify-input/amplify-input.component.ts [hunk 1]
```diff
@@ -26,9 +30,25 @@ export class AmplifyInputComponent {
...
+    this.defaultCountryCode = country_code;
+
+    // TODO: consider better default handling mechanisms across frameworks
+    if (this.isTelInput()) {
+      this.stateMachine.send({
+        type: 'CHANGE',
+        data: { name: 'country_code', value: country_code },
+      });
+    }
+  }
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives/amplify-input/amplify-input.component.ts:50]: Dittoing the implementation from React: https://github.com/aws-amplify/amplify-ui/blob/36ce52646f455e6a3f2d7e9befb4f808efc1061b/packages/react/src/components/Authenticator/shared/UserNameAlias.tsx#L30-L36 We're looking to improve this side effect pattern in a separate pr (Issue #404, [Asana](https://app.asana.com/0/1200334286823427/1201022646170709)). Leaving this as now so I can get started on parity styling 👍

File: packages/angular/projects/ui-angular/src/lib/components/amplify-confirm-sign-up/amplify-confirm-sign-up.component.html [hunk 2]
```diff
@@ -8,10 +8,11 @@
 <ng-template #confirmSignUpForm>
   <form data-amplify-form (submit)="onSubmit($event)" (input)="onInput($event)">
     <fieldset data-amplify-fieldset [disabled]="isPending">
-      <amplify-user-name-alias
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/components/amplify-confirm-sign-up/amplify-confirm-sign-up.component.html:11]: I'm scoping `amplify-user-name-alias` component usage to sign in and sign up only to keep other components clean from country code handling.

File: packages/angular/projects/ui-angular/src/lib/primitives/amplify-input/amplify-input.component.html [hunk 3]
```diff
@@ -1,3 +1,13 @@
+<!-- Phone number field -->
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives/amplify-input/amplify-input.component.html:1]:
```suggestion
<!-- Country code field -->
```

File: packages/angular/projects/ui-angular/src/lib/primitives/amplify-input/amplify-input.component.ts [hunk 4]
```diff
@@ -1,22 +1,26 @@
...
 import { StateMachineService } from '../../services/state-machine.service';
 
 /**
- * Contains an input element and its label. Intended to be used with
- * Angular Reactive Form
+ * Input interface opinionated for authenticator usage.
+ *
+ * TODO: Separate this component out to two parts -- 1) amplify-auth-input that
+ * contains authenticator opionated logic and 2) amplify-base-input primitive
+ * that does not make any auth-related inference.
```
Comment (wlee221) [inline packages/angular/projects/ui-angular/src/lib/primitives/amplify-input/amplify-input.component.ts:17]:
```suggestion
 * TODO: Separate this component out to two parts -- 1) amplify-auth-input that
 * contains authenticator opinionated logic and 2) amplify-text-field primitive
 * that does not make any auth-related inference.
```

File: packages/angular/projects/ui-angular/src/lib/components/amplify-confirm-sign-up/amplify-confirm-sign-up.component.html (hunk 2 above)
Comment (ErikCH) [reply packages/angular/projects/ui-angular/src/lib/components/amplify-confirm-sign-up/amplify-confirm-sign-up.component.html:11]: Good idea!

**Review by ErikCH — APPROVED** [review]
👍

**Pull request merged**